File: leetcode/75. Sort Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        clrs = [0] * 3

        for i in range(len(nums)):
            clrs[nums[i]] += 1

        pos = 0
        for i in range(len(clrs)):
            for j in range(pos, pos + clrs[i]):
                pos += 1
                nums[j] = i

        return nums

File: leetcode/75. Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int h[3];
        for (int i = 0; i<3; i++) {
            h[i] = 0;
        }

        for (int i = 0; i < nums.size(); i++) {
            h[nums[i]] +=1;
        }

        int pos = 0;
        for (int i = 0; i < 3; i++) {
            for (int j = pos; j < pos + h[i]; j++) {
                pos +=1;
                nums.at(j) = i;
            }
        }

        return;

    }
};